services/src/main/java/org/apache/unomi/services/impl/rules/RulesServiceImpl.java

```diff
@@ -381,15 +413,23 @@ public void removeRule(String ruleId) {
         persistenceService.remove(ruleId, Rule.class);
     }
 
+    @Override
+    public void setSetting(String fieldName, Object value) throws NoSuchFieldException, IllegalAccessException {
+        Field field = this.getClass().getDeclaredField(fieldName);
+        field.set(this, value);
+    }
+
+    @Override
+    public Object getSetting(String fieldName) throws NoSuchFieldException, IllegalAccessException {
+        Field field = this.getClass().getDeclaredField(fieldName);
+        return field.get(this);
+    }
+
```

Comment:
This is actually reproducing something we already have at the PersistenceService level (similar pattern). It is solely done to be able to use it in integration tests and the idea was to avoid polluting the service API with setters. Another of implementing this would be to manually set the value by only allowing specific settings. I'm not sure what's safer at that level.

Reply:
Another (cleaner) solution would be do to everything through the OSGi ConfigAdmin but this would require improving the service implementation to support dynamic configuration changes.

Reply:
I will implement the (cleaner) solution of using the OSGi configuration admin.